src/patternfly/components/Menu/menu.scss

```diff
@@ -0,0 +1,294 @@
+  --pf-c-menu__item--PaddingLeft: var(--pf-global--spacer--md);
+  --pf-c-menu__item--FontSize: var(--pf-global--FontSize--md);
+  --pf-c-menu__item--FontWeight: var(--pf-global--FontWeight--normal);
+  --pf-c-menu__item--LineHeight: var(--pf-global--LineHeight--md);
+  --pf-c-menu__item--disabled--Color: var(--pf-global--Color--dark-200);
+
+  // group
+  --pf-c-menu__group-title--PaddingTop: var(--pf-c-menu__item--PaddingTop);
+  --pf-c-menu__group-title--PaddingRight: var(--pf-c-menu__item--PaddingRight);
+  --pf-c-menu__group-title--PaddingBottom: var(--pf-c-menu__item--PaddingBottom);
+  --pf-c-menu__group-title--PaddingLeft: var(--pf-c-menu__item--PaddingLeft);
+  --pf-c-menu__group-title--FontSize: var(--pf-global--FontSize--sm);
+  --pf-c-menu__group-title--FontWeight: var(--pf-global--FontWeight--semi-bold);
+  --pf-c-menu__group-title--Color: var(--pf-global--Color--dark-200);
+
+  // description
+  --pf-c-menu__item-description--FontSize: var(--pf-global--FontSize--xs);
+  --pf-c-menu__item-description--Color: var(--pf-global--Color--200);
+
+  // icon
+  --pf-c-menu__item-icon--MarginRight: var(--pf-global--spacer--sm);
+
+  // toggle icon
+  --pf-c-menu__item-toggle-icon--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-menu__item-toggle-icon--PaddingLeft: var(--pf-global--spacer--sm);
+
+  // select icon
+  --pf-c-menu__item-select-icon--MarginLeft: var(--pf-global--spacer--sm);
+  --pf-c-menu__item-select-icon--Color: var(--pf-global--active-color--100);
+  --pf-c-menu__item-select-icon--FontSize: var(--pf-global--icon--FontSize--sm);
+
+  // external icon
+  --pf-c-menu__item-external-icon--MarginLeft: var(--pf-global--spacer--sm);
+  --pf-c-menu__item-external-icon--Color: var(--pf-global--link--Color);
+  --pf-c-menu__item-external-icon--FontSize: var(--pf-global--icon--FontSize--sm);
+
+  // action
+  --pf-c-menu__item-action--PaddingTop: var(--pf-global--spacer--sm);
+  --pf-c-menu__item-action--PaddingRight: var(--pf-global--spacer--md);
+  --pf-c-menu__item-action--PaddingBottom: var(--pf-global--spacer--sm);
+  --pf-c-menu__item-action--PaddingLeft: var(--pf-global--spacer--md);
+  --pf-c-menu__item-action-icon--Color: var(--pf-global--Color--dark-200);
+  --pf-c-menu__item-action-icon--Height: calc(var(--pf-c-menu__item--FontSize) * var(--pf-c-menu__item--LineHeight));
+  --pf-c-menu__item-action--hover__icon--Color: var(--pf-global--Color--dark-100);
+  --pf-c-menu__item-action--m-favorite__icon--Color: var(--pf-global--disabled-color--200);
+  --pf-c-menu__item-action--m-favorite__icon--FontSize: var(--pf-global--icon--FontSize--sm);
+  --pf-c-menu__item-action--m-favorite--m-favorited__icon--Color: var(--pf-global--palette--gold-400);
+
+  padding-top: var(--pf-c-menu--PaddingTop);
+  padding-bottom: var(--pf-c-menu--PaddingBottom);
+  background-color: var(--pf-c-menu--BackgroundColor);
+  box-shadow: var(--pf-c-menu--BoxShadow);
+
+  &.pf-m-flyout {
```

Comment:
I don't know if it's expensive to be adding and removing things from the DOM to hide/show the submenus but another approach to how this is built could be to render all of the markup and we hide with `display: none; visibility: hidden;` and react just toggles a class (`.pf-m-expanded`) that we then hide/show the menu with via CSS

Reply:
i might be wrong, but i think from @mattnolting and I's conversation, we wanted to reuse `pf-c-menu__item-sublist` for every sublist, and apply `pf-m-flyout` to apply the styling for sublist if it should be positioned like a flyout,
are you suggesting to add `pf-m-flyout` to each `pf-c-menu__item-sublist` instead ? and have a `pf-m-expanded` class if its expanded otherwise its hidden ?